esp_bsp_generic/include/bsp/esp_bsp_generic.h

```diff
@@ -0,0 +1,211 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief ESP BSP: Generic
+ */
+
+#pragma once
+
+#include "sdkconfig.h"
+#include "driver/gpio.h"
+#include "driver/i2c.h"
+#include "driver/sdmmc_host.h"
+#include "iot_button.h"
+
+/**************************************************************************************************
+ *  Pinout
+ **************************************************************************************************/
+/* I2C */
+#define BSP_I2C_SCL      (CONFIG_BSP_I2C_GPIO_SCL)
+#define BSP_I2C_SDA      (CONFIG_BSP_I2C_GPIO_SDA)
+
+/* USB */
+#if ESP_ROM_USB_OTG_NUM > 0
+#define BSP_USB_POS      (USBPHY_DP_NUM)
+#define BSP_USB_NEG      (USBPHY_DM_NUM)
+#else
+#define BSP_USB_POS      (GPIO_NUM_NC)
+#define BSP_USB_NEG      (GPIO_NUM_NC)
+#endif
+
+/* SD card */
+#define BSP_SD_CMD       (CONFIG_BSP_SD_CMD)
+#define BSP_SD_CLK       (CONFIG_BSP_SD_CLK)
+#define BSP_SD_D0        (CONFIG_BSP_SD_D0)
+#define BSP_SD_D1        (CONFIG_BSP_SD_D1)
+#define BSP_SD_D2        (CONFIG_BSP_SD_D2)
+#define BSP_SD_D3        (CONFIG_BSP_SD_D3)
+
+/* Buttons */
+#define BSP_BUTTON_1_IO  (CONFIG_BSP_BUTTON_1_GPIO)
+#define BSP_BUTTON_2_IO  (CONFIG_BSP_BUTTON_2_GPIO)
+#define BSP_BUTTON_3_IO  (CONFIG_BSP_BUTTON_3_GPIO)
+#define BSP_BUTTON_4_IO  (CONFIG_BSP_BUTTON_4_GPIO)
+#define BSP_BUTTON_5_IO  (CONFIG_BSP_BUTTON_5_GPIO)
+
+/* Buttons */
+typedef enum {
+#if CONFIG_BSP_BUTTONS_NUM > 0
+    BSP_BUTTON_1,
+#endif
+#if CONFIG_BSP_BUTTONS_NUM > 1
+    BSP_BUTTON_2,
+#endif
+#if CONFIG_BSP_BUTTONS_NUM > 2
+    BSP_BUTTON_3,
+#endif
+#if CONFIG_BSP_BUTTONS_NUM > 3
+    BSP_BUTTON_4,
+#endif
+#if CONFIG_BSP_BUTTONS_NUM > 4
+    BSP_BUTTON_5,
+#endif
+    BSP_BUTTON_NUM
+} bsp_button_t;
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**************************************************************************************************
+ *
+ * I2C interface
+ *
+ *
+ * After initialization of I2C, use BSP_I2C_NUM macro when creating I2C devices drivers ie.:
+ * \code{.c}
+ * es8311_handle_t es8311_dev = es8311_create(BSP_I2C_NUM, ES8311_ADDRRES_0);
+ * \endcode
```

Comment:
Could we delete this? This is not the recommended way to interface audio codecs now